week-3/Homework/mandatory/2-bush-berries.js

```diff
@@ -10,14 +10,18 @@
   Use the tests to confirm which message to return
 */
 
-function bushChecker() {
-
+function bushChecker(pinkColor) {
+  let bePink = pinkColor.every(item => item == "pink");
+  if (bePink == true) {
```

Comment:
This is absolutely correct (well done!), but the `== true` is actually unnecessary. To see why, let's play computer:

1. The computer runs the `bushes.every(...)` code
2. This evaluates to the boolean `true`
3. This value is then assigned to the `bePink` variable
4. The computer reads the `if` statement and sees that the comparison is `bePink == true`
5. It evaluates `bePink == true`
6. The `bePink` variable is assigned to `true` so the expression becomes `true == true`
7. This evaluates to a single `true` value (because `true` is equal to `true`!)
8. The `if` statement runs the body (returning `"Bush is safe to eat from"`) because the value is `true`

If we changed the code to just `if (bePink)` we could skip steps 5 - 7! This is very common when writing JS.

---

Additionally, in a situation like this where we only have 1 condition in the `if` statement, we usually don't separate it out into a separate variable (`bePink` in this case). Instead we just *inline* the comparison into the `if` statement. Do you think you could change that?

Reply:
Yes I can write like
bePink ? "Bush is safe to eat from" : "Toxic! Leave bush alone!"